src/TestFramework/Factory.php

```diff
@@ -105,19 +106,33 @@ public function create(string $adapterName, bool $skipCoverage): TestFrameworkAd
             );
         }
 
-        if ($adapterName === TestFrameworkTypes::CODECEPTION) {
-            $codeceptionConfigPath = $this->configLocator->locate(TestFrameworkTypes::CODECEPTION);
+        $extensionsClassExists = class_exists('\Infection\ExtensionInstaller\GeneratedExtensionsConfig');
 
-            return CodeceptionAdapterFactory::create(
-                (new TestFrameworkFinder('codecept'))->find(),
-                $this->tmpDir,
-                $codeceptionConfigPath,
-                null,
-                $this->jUnitFilePath,
-                $this->projectDir,
-                $this->infectionConfig->getSourceDirectories(),
-                $skipCoverage
-            );
+        if ($extensionsClassExists) {
```

Comment:
instead of having:

```php
if ($extensionsCLassExists) {
  // ...
}

throw new InvalidArgumentException();
```

You could invert the condition to avoid one nesting level:

```php
if (!$extensionsCLassExists) {
  throw new InvalidArgumentException();
}

// ...
```